### PR TITLE
refactor(experimental): graphql: token-2022 extensions: Reallocate Instruction

### PR DESCRIPTION
**Pull request by Hrushi20**
This PR adds support for Token-2022's Reallocate instruction in the GraphQL schema.

Continuing work on https://github.com/solana-labs/solana-web3.js/issues/2406.

### CODE REVIEW

File: packages/rpc-graphql/src/__tests__/transaction-tests.ts [hunk 1]
```diff
@@ -1480,6 +1480,89 @@ describe('transaction', () => {
...
+                                        account: {
+                                            address: expect.any(String),
+                                        },
+                                        extensionTypes: expect.any([expect.any(String)]),
+                                        multisigOwner: {
+                                            address: expect.any(String),
+                                        },
+                                        owner: {
+                                            address: expect.any(String),
+                                        },
```
Comment (buffalojoec) [inline packages/rpc-graphql/src/__tests__/transaction-tests.ts:1550]: `owner` should be null here.

File: packages/rpc-graphql/src/schema/instruction.ts [hunk 2]
```diff
@@ -670,6 +670,20 @@ export const instructionTypeDefs = /* GraphQL */ `
         signers: [Address]
     }
 
+    """
+    SplToken-2022: Reallocate instruction
+    """
+    type SplTokenReallocate implements TransactionInstruction {
+        programId: Address
+        account: Account
+        extensionTypes: [String]
```
[the remark this answers is not in the document]
Comment (Hrushi20) [reply packages/rpc-graphql/src/schema/instruction.ts:679]: Thanks. I'll add that as well.

File: packages/rpc-graphql/src/schema/instruction.ts [hunk 3]
```diff
@@ -1093,6 +1093,22 @@ export const instructionTypeDefs = /* GraphQL */ `
         metadata: Account
         updateAuthority: Account
     }
+    """
+    SplToken-2022: Reallocate instruction
+    """
+    type SplTokenReallocate implements TransactionInstruction {
+        programId: Address
+        account: Account
+        extensionTypes: [String]
```
Comment (buffalojoec) [inline packages/rpc-graphql/src/schema/instruction.ts:1102]: Just need to update this field and this one should be good to go.

File: packages/rpc-graphql/src/schema/types.ts [hunk 4]
```diff
@@ -86,4 +86,32 @@ export const typeTypeDefs = /* GraphQL */ `
         BASE_58
         BASE_64
     }
+
+    enum ExtensionType {
```
Comment (buffalojoec) [inline packages/rpc-graphql/src/schema/types.ts:90]: Actually I think you'll also need to define a resolver to go to and from string values here. You'll see the others under `resolvers/types.ts`

**Comment by buffalojoec** [comment]
@Hrushi20 commandeered #2811.

**Pull request closed without merging**